src/workerd/api/urlpattern-standard.c++

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "urlpattern-standard.h"
+
+#include "ada.h"
+
+namespace workerd::api::urlpattern {
+std::optional<URLPattern::URLPatternRegexEngine::regex_type> URLPattern::URLPatternRegexEngine::
+    create_instance(std::string_view pattern, bool ignore_case) {
+  jsg::Lock& js = jsg::Lock::from(v8::Isolate::GetCurrent());
+  jsg::Lock::RegExpFlags flags = jsg::Lock::RegExpFlags::kUNICODE_SETS;
+  if (ignore_case) {
+    flags = static_cast<jsg::Lock::RegExpFlags>(
+        flags | static_cast<int>(jsg::Lock::RegExpFlags::kIGNORE_CASE));
+  }
+
+  return js.tryCatch([&]() -> std::optional<regex_type> {
+    return jsg::JsRef(js, js.regexp(kj::StringPtr(pattern.data(), pattern.size()), flags));
+  }, [&](auto reason) -> std::optional<regex_type> { return std::nullopt; });
+}
+
+bool URLPattern::URLPatternRegexEngine::regex_match(
+    std::string_view input, const regex_type& pattern) {
+  jsg::Lock& js = jsg::Lock::from(v8::Isolate::GetCurrent());
+  return pattern.getHandle(js).match(js, kj::StringPtr(input.data(), input.size()));
+}
+
+std::optional<std::vector<std::optional<std::string>>> URLPattern::URLPatternRegexEngine::
+    regex_search(std::string_view input, const regex_type& pattern) {
+  jsg::Lock& js = jsg::Lock::from(v8::Isolate::GetCurrent());
+  KJ_IF_SOME(matches, pattern.getHandle(js)(js, kj::StringPtr(input.data(), input.size()))) {
+    std::vector<std::optional<std::string>> results;
+    results.reserve(matches.size());
+    for (size_t i = 1; i < matches.size(); i++) {
```

Comment:
Shouldn't this be `<=`?

Reply:
No because regexp exec() method returns the original input as the first argument. We want to NOT include that in the returning vector.

Reply:
Nit: I would add that as a comment